src/os/filestore/JournalingObjectStore.cc

```diff
@@ -81,10 +81,10 @@ int JournalingObjectStore::journal_replay(uint64_t fs_op_seq)
 
     dout(3) << "journal_replay: applying op seq " << seq << dendl;
     bufferlist::iterator p = bl.begin();
-    list<Transaction*> tls;
+    vector<ObjectStore::Transaction> tls;
     while (!p.end()) {
-      Transaction *t = new Transaction(p);
-      tls.push_back(t);
+      Transaction t (p);
+      tls.push_back(std::move(t));
```

Comment:
you might consider using `tls.emplace_back(p);` here to combine the construction with the push_back()